core/os/os2/env_posix.odin

```diff
@@ -0,0 +1,77 @@
+//+private
+//+build darwin, netbsd, freebsd, openbsd
+package os2
+
+import "base:runtime"
+
+import "core:strings"
+import "core:sys/posix"
+
+_lookup_env :: proc(key: string, allocator: runtime.Allocator) -> (value: string, found: bool) {
+	if key == "" {
+		return
+	}
+
+	TEMP_ALLOCATOR_GUARD(ignore=is_temp(allocator))
```

Comment:
Remove the `ignore` check. That is useless in this context because `temp_allocator()` is private the package anyway.

Don't try to repeat the bodge that was in `core:os`.